rust/src/ssh/ssh.rs

```diff
@@ -179,8 +211,16 @@ impl SSHState {
                             let remlen = rem.len() as u32;
                             hdr.record_left = head.pkt_len - 2 - remlen;
                             //header with rem as incomplete data
-                            if head.msg_code == SSH_MSG_NEWKEYS {
-                                hdr.flags = SSHConnectionState::SshStateFinished;
+                            match head.msg_code { 
+                                parser::MessageCode::SshMsgNewKeys => {
+                                    hdr.flags = SSHConnectionState::SshStateFinished;
+                                }
+                                parser::MessageCode::SshMsgKexinit if HASSH_ENABLED.load(Ordering::Relaxed) => {
+                                    // saving type of incomplete kex message
+                                    hdr.record_left_msg = parser::MessageCode::SshMsgKexinit;
+                                    return AppLayerResult::incomplete(SSH_RECORD_HEADER_LEN as u32, hdr.record_left as u32);
```

Comment:
We need to set a limit on `hdr.record_left` here.
Otherwise, Suricata is vulnerable to memory exhaustion as it will store in memory the whole record which can take up to 4 gigabytes (threat with both ssh endpoints controlled by attacker)

Reply:
Is it ok to put a limit of 65535 bytes as a maximum size?

Reply:
Looks good to me

Reply:
Thanks a lot/